src/Baum/Traits/NestedSet.php

```diff
@@ -198,7 +198,7 @@ public function getQualifiedLeftColumnName()
      */
     public function getLeft()
     {
-        return $this->getAttribute($this->getLeftColumnName());
+        return (int) $this->getAttribute($this->getLeftColumnName());
```

Comment:
The function/method destroyDescendants() anticipates that getLeft may return null, and this function/scenario is not covered  in the tests. So your change of casting this to int may cause unexpected functionality. If you want this to be committed you need to check each use of getLeft and ensure no where expects it to return null, and if it does adjust that code accordingly, along with tests to ensure the original functionality is still being met, as if this had a proper test case then your changes would cause it to fail
<img width="880" alt="Screenshot 2024-06-10 at 10 26 54" src="https://github.com/TopOnePercent/baum/assets/28979747/3176f59f-d31a-4584-a151-70c044334182">

Reply:
I would propose to use empty() instead of is_null(). That would cover all falsy values, since it doesn't make sense for nodes to have 0 or null values for the left and right columns (null would be cast to 0 either way in the proposed fix).
Not sure how to cover this with tests atm, though.

Reply:
if null is casted to 0 with (int) we could just check specifically for that if ($this->getRight() == 0 ...

That way we can be exact with what we are checking.

As for tests it should be simple to just run it against a node with no right or left and ensure no error is thrown using a try catch something like

try {
       $node->destroyDescendants();
      $this->assertTrue(TRUE);
    } catch () {
      $this->fail();
    }

Reply:
@BilalAslim please see, if everything is correct now. Thanks!

Reply:
@cliche23 I am working on the review now, hopefully can get it done today or tomorrow.

Reply:
We can defiantly write a better test because this still passes even with the null, and doesn't actually ensure we fixed the problem of avoiding running the other parts of the code. But I tested manually myself and this fix we applied here does keep the code working as it was originally intended.